firebase-appdistribution/test-app/src/main/java/com/googletest/firebase/appdistribution/testapp/NotificationFeedbackTrigger.kt

```diff
@@ -0,0 +1,215 @@
+package com.googletest.firebase.appdistribution.testapp
+
+import android.Manifest.permission.POST_NOTIFICATIONS
+import android.annotation.SuppressLint
+import android.app.*
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager.PERMISSION_GRANTED
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.net.Uri
+import android.os.Build
+import android.os.Bundle
+import android.util.Log
+import androidx.activity.result.ActivityResultCaller
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import androidx.core.content.ContextCompat
+import com.google.firebase.appdistribution.ktx.appDistribution
+import com.google.firebase.ktx.Firebase
+import com.googletest.firebase.appdistribution.testapp.NotificationFeedbackTrigger.SCREENSHOT_FILE_NAME
+import com.googletest.firebase.appdistribution.testapp.NotificationFeedbackTrigger.takeScreenshot
+import java.io.IOException
+
+@SuppressLint("StaticFieldLeak") // Reference to Activity is set to null in onActivityDestroyed
+object NotificationFeedbackTrigger : Application.ActivityLifecycleCallbacks {
+    private const val TAG: String = "NotificationFeedbackTrigger"
+    private const val FEEBACK_NOTIFICATION_CHANNEL_ID = "InAppFeedbackNotification"
+    private const val FEEDBACK_NOTIFICATION_ID = 1
+    const val SCREENSHOT_FILE_NAME =
+        "com.googletest.firebase.appdistribution.testapp.screenshot.png"
+
+    private var isEnabled = false
+    private var hasRequestedPermission = false
+    private var currentActivity: Activity? = null
+    private var requestPermissionLauncher: ActivityResultLauncher<String>? = null
+
+    fun initialize(application: Application) {
+        // Create the NotificationChannel, but only on API 26+ because
+        // the NotificationChannel class is new and not in the support library
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val channel = NotificationChannel(
+                FEEBACK_NOTIFICATION_CHANNEL_ID,
+                application.getString(R.string.feedback_notification_channel_name),
+                NotificationManager.IMPORTANCE_HIGH
```

Comment:
Yes, but unfortunately low importance messages sometimes just don't show up at all 😬  So I decided to err on the side of visibility - making sure the notification is seen. Since this is sample code selecting the appropriate importance is ultimately up to the developers integrating code based on this example into their app.

Reply:
Got it. I think it could be worth trying to make sure there's a level that makes sense though. If all of the levels either 1) make a sound or pop down over the current view, or 2) show up inconsistently, then it might not make sense to even supply sample code for a notification trigger.

Reply:
I guess it's up to the developer even if it makes sounds. Some developers might really want to get feedback 😄